disco_aws_automation/disco_elasticsearch_archive.py

```diff
@@ -207,34 +225,39 @@ def _archivable_indices(self):
         yesterday = datetime.date.today() - datetime.timedelta(days=1)
         yesterday_str = yesterday.strftime('%Y.%m.%d')
 
-        return [index[0]
-                for index in self._get_all_indices_and_sizes()
-                if index[0][-10:] < yesterday_str]
+        return [index['index']
+                for index in self._get_all_indices_stats()
+                if index['index'][-10:] < yesterday_str]
 
     def _all_index_names(self):
         """ Return list of all indices """
-        return [index[0]
-                for index in self._get_all_indices_and_sizes()]
+        return [index['index']
+                for index in self._get_all_indices_stats()]
 
-    def _get_all_indices_and_sizes(self):
+    def _get_all_indices_stats(self):
         """
-        Return list of all the indices and their sizes sorted by the date from
-        oldest to latest
+        Return list of all the indices, their sizes and their numbers of shards
+        sorted by the date from oldest to latest
         """
-        stats = self.es_client.indices.stats()
+        cat_stats = self.es_client.cat.indices(bytes='b', h='index,store.size,pri,rep')
```

Comment:
Is this cat same as /_cat? I was under the impression those endpoints were for humans and shell scripts. Is it any different to /_stats